### ISSUE TITLE
How to retrieve matching docs

### ISSUE THREAD

**Issue by ejeh** [sebelga/gstore-node#151]
Pls how can I retrieve or count matching documents from datastore collection using gstore-node module like the count method used in moogodb @sebelga . Thanks in advance genius.

**Comment by sebelga** [comment]
Hello,
Are you saying the total matching of entities in a query or the total entities in the Datastore?
The latter has been discussed here: https://github.com/sebelga/gstore-node/issues/126 but hasn't been implemented yet. And for the matching document in a query, it would be to check the length of the `entities` returned by the query.

**Comment by ejeh** [comment]
> Hello,
> Are you saying the total matching of entities in a query or the total entities in the Datastore?
> The latter has been discussed here: #126 but hasn't been implemented yet. And for the matching document in a query, it would be to check the length of the `entities` returned by the query.

**Comment by ejeh** [comment]
@sebelga, in simpler terms, I want to return something if a value finds a matching attribute in a datastore entity. Sorry, my first question is a bit confusing. Thanks, bro.

**Comment by ejeh** [comment]
Pls, another question @sebelga. Is it only in the current version of gstore-node that the  .populate() method is applicable? If that's the case the version keeps throwing this "TypeError: Cannot read property 'Schema' of undefined"

**Comment by ejeh** [comment]
pls, what I'm I not doing right?

model.js

const { instances } = require("gstore-node");
const gstore = instances.get("unique-id");
const Schema = gstore.Schema;

const AdminSchema = new Schema(
  {
    nonce: {
      type: Number,
      default: randomNonce(),
      required: [true, "Why no authentication nonce?"]
    },
    publicAddress: {
      type: String,
      max: 42,
      required: [false, "Why no MetaMask address?"]
    }
)

server.js

const app = express();

const gstore = require("gstore-node");
const { Datastore } = require("@google-cloud/datastore");

const datastore = new Datastore({
  projectId: "bezopdb"
});

gstore.connect(datastore);

Here is the error i get
Users/godfreyejeh/Documents/marketplace-backend-new copy/src/api/admin/model.js:47
var Schema = gstore.Schema;
                    ^

TypeError: Cannot read property 'Schema' of undefined
    at Object.<anonymous> (/Users/godfreyejeh/Documents/marketplace-backend-new copy/src/api/admin/model.js:37:23)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at loader (/Users/godfreyejeh/Documents/marketplace-backend-new copy/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/godfreyejeh/Documents/marketplace-backend-new copy/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/Users/godfreyejeh/Documents/marketplace-backend-new copy/src/api/auth/controller.js:12:1)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at loader (/Users/godfreyejeh/Documents/marketplace-backend-new copy/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/godfreyejeh/Documents/marketplace-backend-new copy/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)

**Comment by sebelga** [comment]
Hello,

Sorry, maybe the documentation hasn't been updated everywhere since the last major release. You need to access the same instance of gstore in the different files.

```js
// server.js

const app = express();

const { Gstore, instances }  = require("gstore-node");
const { Datastore } = require("@google-cloud/datastore");

const datastore = new Datastore({ projectId: "bezopdb" });
const gstore = new Gstore();
gstore.connect(datastore);

instances.set('unique-id', gstore);
```

```js
// models.js (and any other files)

const { instances } = require("gstore-node");
const gstore = instances.get("unique-id");
const Schema = gstore.Schema;

...
```

Can you tell me if the documentation is outdated?

**Comment by ejeh** [comment]
The documentation is not outdated. I did the exact same thing you just put up but still can't get it to work in a bigger project some level deeper. Works fine in a very smaller project though.  Just wondering why I cannot access the gstore instance in a project some level deeper. for now, I had to connect to datastore in the model file where my schema is. Thanks.

**Comment by sebelga** [comment]
It sounds like you have a problem of order of import. I would suggest to create a file just for the Gstore initialization and import it from there.

```js
// gstore.js

const { Datastore } = require('@google-cloud/datastore');
const { Gstore } = require('gstore-node');

const datastore = new Datastore();
const gstore = new Gstore();
gstore.connect(datastore);

module.exports = { gstore };
```

**Comment by ejeh** [comment]
Thanks a million!!

**Comment by ejeh** [comment]
I just hope I can use the library to get everything I needed to interact with datastore done...
